src/devices_models/devices/common/cost_functions.jl

```diff
@@ -182,18 +182,25 @@ function _pwlgencost_sos(
     psi_container::PSIContainer,
     variable::JV,
     cost_component::Vector{NTuple{2, Float64}},
+    status::Union{Nothing, JV} = nothing,
```

Comment:
This implementation still troubles me. It needs to be refactored to something more elegant. For now we can leave it.

Reply:
I don't need that to be optional arg, I'll remove the Union and keep it only JV

Reply:
I take that back, only case when you don't have status variable is when your using pwl cost for reserves